src/app/utils/__tests__/date.spec.ts

```diff
@@ -67,4 +71,69 @@ describe('Date Util', () => {
       })
     })
   })
+
+  describe('getStartOfDay', () => {
+    // Default the mock time to bangkok time
+    const MOCK_TIME = '2023-06-22T23:30:00-07:00'
+    beforeAll(() => {
+      // Mock the return value for new Date() to avoid flaky test
+      jest.useFakeTimers()
+      jest.setSystemTime(new Date(MOCK_TIME))
+    })
+
+    afterAll(() => {
+      jest.useRealTimers()
+    })
+
+    it(`should return today's start of day in Singapore timezone if option is not given`, () => {
+      // Act
+      const result = getStartOfDay()
+      // Assert
+      const expected = new Date('2023-06-23T00:00:00+08:00')
+      expect(result).toEqual(expected)
+    })
+
+    it(`should return today's start of day in Singapore timezone if date and timezone is not given`, () => {
+      // Act
+      const result = getStartOfDay({})
+      // Assert
+      const expected = new Date('2023-06-23T00:00:00+08:00')
+      expect(result).toEqual(expected)
+    })
+
+    it(`should return start of day in Singapore timezone if timezone is not given`, () => {
+      // Act
+      const date = new Date('2020-04-13T05:00:00.000+20:00')
+      const result = getStartOfDay({ date: date })
+      // Assert
+      const expected = new Date(date.setUTCHours(-8, 0, 0, 0))
```

Comment:
Since we are hard-coding the input date, I think this should be hard-coded as well I think. Doesn't make sense when we do computations in the function and similarly compute stuff in the tests.

Reply:
Anyway, a note for next time as well! Can just use `date.setUTCHours(-8)` and no need the other args as they are optional haha based on the documentation [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/setUTCHours).

Reply:
i shouldn't hardcode this, i missed it! updated it